splot/_viz_esda_mpl.py

```diff
@@ -1062,3 +1160,113 @@ def moran_loc_bv_scatterplot(moran_loc_bv, p=None,
         ax.plot(lag, fit.predy, **fitline_kwds)
         ax.scatter(moran_loc_bv.zy, fit.predy, **scatter_kwds)
     return fig, ax
+
+
+def moran_facette(moran_matrix, figsize=(16,12),
+                  scatter_bv_kwds=None, fitline_bv_kwds=None,
+                  scatter_glob_kwds=dict(color='#737373'), fitline_glob_kwds=None):
+    """
+    Moran Facette visualization.
+    Includes BV Morans and Global Morans on the diagonal.
+    
+    Parameters
+    ----------
+    moran_matrix : esda.moran.Moran_BV_matrix instance
+        Dictionary of Moran_BV objects
+    figsize : tuple, optional
+        W, h of figure. Default =(16,12)
+    scatter_bv_kwds : keyword arguments, optional
+        Keywords used for creating and designing the scatter points of
+        off-diagonal Moran_BV plots.
+        Default =None.
+    fitline_bv_kwds : keyword arguments, optional
+        Keywords used for creating and designing the moran fitline of
+        off-diagonal Moran_BV plots.
+        Default =None.
+    scatter_glob_kwds : keyword arguments, optional
+        Keywords used for creating and designing the scatter points of
+        diagonal Moran plots.
+        Default =None.
+    fitline_glob_kwds : keyword arguments, optional
+        Keywords used for creating and designing the moran fitline of
+        diagonal Moran plots.
+        Default =None.
+
+    Returns
+    -------
+    fig : Matplotlib Figure instance
+        Bivariate Moran Local scatterplot figure
+    axarr : matplotlib Axes instance
+        Axes in which the figure is plotted
+
+    Examples
+    --------
+    Imports
+    
+    >>> import matplotlib.pyplot as plt
+    >>> import libpysal.api as lp
+    >>> import numpy as np
+    >>> from esda.moran import Moran_BV_matrix
+    >>> from splot.esda import moran_facette
+    
+    Load data and calculate Moran Local statistics
+    
+    >>> f = lp.open(lp.get_path("sids2.dbf"))
+    >>> varnames = ['SIDR74',  'SIDR79',  'NWR74',  'NWR79']
+    >>> vars = [np.array(f.by_col[var]) for var in varnames]
+    >>> w = lp.open(lp.get_path("sids2.gal")).read()
+    >>> moran_matrix = Moran_BV_matrix(vars,  w,  varnames = varnames)
+    
+    Plot
+    
+    >>> fig, axarr = moran_facette(moran_matrix)
+    >>> plt.show()
+    
+    Customize plot
+    
+    >>> fig, axarr = moran_facette(moran_matrix,
+    ...                            fitline_bv_kwds=dict(color='#4393c3'))
+    >>> plt.show()
+    
+    """
+    nrows = int(np.sqrt(len(moran_matrix))) + 1
+    ncols = nrows
+    
+    fig, axarr = plt.subplots(nrows, ncols, figsize=figsize,
+                              sharey=True, sharex=True)
+    fig.suptitle('Moran Facette')
+    
+    for row in range(nrows):
```

Comment:
Just as a suggestion: would it make sense to either use [`seaborn.FacetGrid`](https://seaborn.pydata.org/generated/seaborn.FacetGrid.html#seaborn.FacetGrid) to make our facets more abstract, that way making them more re-usable across the library? You might have considered this already and it might make not sense but just checking here. Ideally, if we're gonna have facets in several places of `splot`, the more we can reuse, the easier it'll be to maintain.

Reply:
`seaboard.FacetGrid` takes as an input ```data : DataFrame
Tidy (“long-form”) dataframe where each column is a variable and each row is an observation.```

Since the `Moran_BV_matrix` returns `Moran_BV` objects I thought reusing the scatterplot, that plots `Moran_BV` objects is the way to go. Otherwise we would need a `Moran_Matrix` that returns spatially lagged values and values.